### PR TITLE
Evaluate win rate for loaded SGFs in the background

### PR DESCRIPTION
**Pull request by tommadams**
This PR adds a map of auxiliary information for positions played during a game (and its variations) to the GtpPlayer. This gives us somewhere to put extra data for moves played without making the base MctsNodes any bigger, since there are normally many mode MctsNodes in a search tree than moves played.

The GtpPlayer maintains a queue of nodes to perform win-rate evaluation on. The queue holds all nodes in the current variation that have had fewer than num_eval_reads win-rate evaluation tree searches performed on them. The number of tree searches performed for win-rate evaluation is tracked separately that the tree search visit count to ensure fairness.

### CODE REVIEW

File: minigui/winrate_graph.ts [hunk 1]
```diff
@@ -45,10 +45,11 @@ class WinrateGraph extends View {
   protected mainLine: number[] = [];
   protected variation: number[] = [];
 
-  protected moveNum = 0;
-
   protected xScale = MIN_POINTS;
```
Comment (amj) [inline minigui/winrate_graph.ts]: is this xScale unneeded, now?

File: minigui/winrate_graph.ts [hunk 2]
```diff
@@ -191,13 +202,16 @@ class WinrateGraph extends View {
...
     ctx.textAlign = 'left';
     ctx.fillStyle = '#ffe';
-    let y = 0;
-    let values = this.variation.length > 0 ? this.variation : this.mainLine;
+    let q = 0;
+    let values =
+        this.activePosition.isMainLine ? this.mainLine : this.variation;
     if (values.length > 0) {
-      y = values[this.moveNum];
+      console.log(values);
```
Comment (amj) [inline minigui/winrate_graph.ts]: obsolete?

File: cc/gtp_player.h [hunk 3]
```diff
@@ -47,18 +48,52 @@ class GtpPlayer : public MctsPlayer {
 
     // If true, we will always pass if the opponent passes.
     bool courtesy_pass = false;
+
+    // Number of times to perform tree search for each position in an SGF in
+    // order to evalute a win rate estimation.
+    int num_eval_reads = 4;
```
Comment (amj) [inline cc/gtp_player.h]: can there be a GTP command (or a flag) for this?

File: cc/gtp_player.cc [hunk 4]
```diff
@@ -115,13 +116,55 @@ void GtpPlayer::Run() {
...
+}
+
 Coord GtpPlayer::SuggestMove() {
   if (courtesy_pass_ && root()->move == Coord::kPass) {
     return Coord::kPass;
   }
   return MctsPlayer::SuggestMove();
 }
 
+bool GtpPlayer::PlayMove(Coord c) {
```
Comment (amj) [inline cc/gtp_player.cc]: can you comment a bit about how this is different than genmove / make explicit the linkages here w.r.t. the front end?

Comment (tommadams) [reply cc/gtp_player.cc]: I've added a TODO to gtp_player.h to document all the Handle* methods of GtpPlayer. I'll do a pass over the comments & documentation once the important features are mostly finished

File: cc/gtp_player.h (hunk 3 above)
Comment (tommadams) [reply cc/gtp_player.h]: added a "ponder winrate N" command

File: minigui/winrate_graph.ts (hunk 2 above)
Comment (tommadams) [reply minigui/winrate_graph.ts]: removed.

File: minigui/winrate_graph.ts (hunk 1 above)
Comment (tommadams) [reply minigui/winrate_graph.ts]: added a comment describing why xScale is needed

**Pull request merged**